### ISSUE TITLE
deploy logging stack as early as possible in shoot reconciliation flow

### ISSUE THREAD

**Issue by vlerenc** [gardener/gardener#2457]
**How to categorize this issue?**
/area control-plane logging ops-productivity
/kind enhancement
/priority normal

**What would you like to be added**:
I checked a cluster that couldn't be deleted because of infra dependencies:
```
$ k get pods
NAME                                         READY   STATUS              RESTARTS   AGE
daily-curator-1592268360-fgw25               0/1     Completed           0          137m
elasticsearch-logging-0                      2/2     Running             0          18d
etcd-events-0                                2/2     Running             0          18d
etcd-main-0                                  2/2     Running             0          18d
gardener-resource-manager-549878d987-mkskc   1/1     Running             0          11h
hourly-curator-1592275260-cvzb6              0/1     Completed           0          22m
kibana-logging-5f44699c95-txjzs              2/2     Running             0          18d
kube-controller-manager-9f655d794-lgkpg      0/1     ContainerCreating   0          2d16h
kube-scheduler-76f7bff879-h5hcw              0/1     CrashLoopBackOff    748        39h
term-6224465904394368752                     1/1     Running             0          8m12s
```

What I noticed was:
1. There was no API server anymore, but dependent components. This is more cosmetic as it creates somewhat needless additional confusion to ops/support staff. The expectation is that the components get deleted in reverse order of creation.
2. I couldn't find the TF pod logs in the logging stack, but maybe I was looking wrong? The only thing I saw were scheduler logs. Where are the TF pod logs? Dropped after the TF pod issue (constant retries that flooded fluentd that then dropped all else)?

**Why is this needed**:
(1) is about less noise/avoidable errors and (2) about finding the logs for what the cluster operation error refers to.

**Comment by ialidzhikov** [comment]
> 2. I couldn't find the TF pod logs in the logging stack, but maybe I was looking wrong? The only thing I saw were scheduler logs. Where are the TF pod logs? Dropped after the TF pod issue (constant retries that flooded fluentd that then dropped all else)?

Actually the provider extensions log the full content of the logs of a terraform Pod after its execution. See https://github.com/gardener/gardener/blob/a3c0c1f06848b44e8492319c83e51b8f99cd3901/extensions/pkg/terraformer/terraformer.go#L204-L212

When I search for logs of a terraformer Pod, I go to the seed Kibana (where the logs from non-Shoot namespaces go, incl. the logs of the extension controllers) and I filter the logs by the extension Pod name as `kubernetes.pod_name,is,gardener-extension-provider-<provider>` and search for `"<shoot-name>"`.

**Comment by vlerenc** [comment]
Thanks @ialidzhikov.

Out of curiosity: We have had issues some weeks ago with TF logs flooding fluentd that couldn't be put into the logging stack, because it was gone, wasn't that the case?

Overall: Having to collect control-plane-local and seed-extension-global logs to piece together the view of a cluster makes trouble-shooting a cluster harder. Then again, that's nothing really new.

**Comment by rfranzke** [comment]
> There was no API server anymore, but dependent components. This is more cosmetic as it creates somewhat needless additional confusion to ops/support staff. The expectation is that the components get deleted in reverse order of creation.

I think we decided to do this on purpose in the early days of Gardener: We never explicitly deleted the other components to "save the code" and we relied on the namespace controller that cleans everything when we delete the shoot namespace. Do you suggest to change this now?

> I couldn't find the TF pod logs in the logging stack, but maybe I was looking wrong? The only thing I saw were scheduler logs. Where are the TF pod logs? Dropped after the TF pod issue (constant retries that flooded fluentd that then dropped all else)?

Can you share the `Shoot` spec? With #2320 logs are discarded in case the shoot's purpose is `testing` (or if hibernated, but that seems to not be the case according to what you posted in the description).

**Comment by rfranzke** [comment]
/title Deploy logging stack as early as possible in shoot reconciliation flow

for now :)

**Comment by vpnachev** [comment]
Fixed by #2750 
/close
